src/EventType.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+/**
+ * This enum represents all the possible types of events that a Sentry server
+ * supports.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class EventType
+{
+    /**
+     * @var string The value of the enum instance
+     */
+    private $value;
+
+    /**
+     * @var array<string, self> A list of cached enum instances
+     */
+    private static $instances = [];
+
+    private function __construct(string $value)
+    {
+        $this->value = $value;
+    }
+
+    /**
+     * Creates an instance of this enum for the "default" value.
+     */
+    public static function default(): self
+    {
+        return self::getInstance('default');
+    }
+
+    /**
+     * Creates an instance of this enum for the "transaction" value.
+     */
+    public static function transaction(): self
+    {
+        return self::getInstance('transaction');
+    }
+
+    public function __toString(): string
+    {
+        return $this->value;
+    }
+
+    private static function getInstance(string $value): self
```

Comment:
Because this enum (and I will refactor the old ones too in the near future) is now implemented using the flyweight pattern so that you always get the same instance for the same value, which means improved memory performances and allows comparing it using `===`